plugins/restful/RestfulBase.php

```diff
@@ -815,7 +815,16 @@ public function process($path = '', array $request = array(), $method = \Restful
       $this->getRateLimitManager()->checkRateLimit($request);
     }
 
-    return $this->{$method_name}($path);
+    $return = $this->{$method_name}($path);
+
+    if (empty($request['__application']['rest_call'])) {
```

Comment:
We are always switching the user when doing a `setAccount`, but we are only switching it back if the call is from cURL?